### ISSUE TITLE
no way to do tcp flow control

### ISSUE THREAD

**Issue by deanhiller** [AsyncHttpClient/async-http-client#177]
We are receiving http chunked response from 3 streams.  Naturally we do not want one stream to get ahead of another or we can get out of memory.  To slow down one of the streams, the correct way is to use tcp flow control which means we need async-http-client to NOT read so the nic buffer fills up and tcp flow control notifies down stream to stop sending packets (I have tested this with another nio library at one point and it works great).

So basically, async-http-client should be first calling a listener.hasData method and we return true or false on whether we want the data to be read on that socket.  Without this we are getting outOfMemory errors from one server being too fast and others being slower.
